src/formatting_stack/formatters/clean_ns.clj

```diff
@@ -57,20 +53,42 @@
 (def default-namespaces-that-should-never-cleaned
   #{'user 'dev})
 
+(defn replaceable-ns-form
+  [{:keys [how-to-ns-opts refactor-nrepl-opts namespaces-that-should-never-cleaned libspec-whitelist]} filename]
+  (when (and (try-require filename)
+             (not (impl/has-duplicate-requires? filename)))
+    (util.ns/replaceable-ns-form
+     filename
+     (make-cleaner how-to-ns-opts refactor-nrepl-opts namespaces-that-should-never-cleaned libspec-whitelist filename)
+     how-to-ns-opts)))
+
 (defn format!
-  [{:keys [how-to-ns-opts refactor-nrepl-opts namespaces-that-should-never-cleaned libspec-whitelist]}
-   files]
+  [this files]
   (->> files
        (process-in-parallel! (fn [filename]
-                               (when (and (try-require filename)
-                                          (not (impl/has-duplicate-requires? filename)))
-                                 (clean! how-to-ns-opts
-                                         refactor-nrepl-opts
-                                         namespaces-that-should-never-cleaned
-                                         libspec-whitelist
-                                         filename)))))
+                               (when-let [ns-replacement (replaceable-ns-form this filename)]
+                                 (println "Cleaning unused imports:" filename)
+                                 (write-ns-replacement! filename ns-replacement)))))
   nil)
 
+(defn lint! [this files]
+  (->> files
+       (process-in-parallel! (fn [filename]
+                               (when-let [{:keys [final-ns-form-str
+                                                  original-ns-form-str]} (replaceable-ns-form this filename)]
+                                 (let [diff (diff/unified-diff filename original-ns-form-str final-ns-form-str)]
+                                   (->> (diff->line-numbers diff)
+                                        (mapv (fn [{:keys [start]}]
+                                                {:filename filename
+                                                 :diff diff
+                                                 :level :warning
+                                                 :column 0
+                                                 :line start
+                                                 :msg "ns can be cleaned"
```

Comment:
```suggestion
                                                 :msg "ns has unused imports and can be cleaned"
```

Reply:
The ns _might_ have unused imports, but it could also be unused requires, refers or be (partly) unordered. I struggle to find a good description of everything for this message.

Reply:
Check. I was basing my suggestion on the `println` in the `format!` function. Maybe my point is more to at least have the two consistent :)

But yeah, I see how my suggestion is misleading/painting an incomplete picture as well.